### ISSUE TITLE
Painting a RecordingSurface to an SVGSurface with offset doesn't work

### ISSUE THREAD

**Issue by Evpok** [Kozea/cairocffi#99]
Executing the following code

```python
import cairocffi as cairo

def test1():
    aux = cairo.RecordingSurface(cairo.CONTENT_COLOR_ALPHA, None)
    aux_context = cairo.Context(aux)
    aux_context.move_to(0, 0)
    aux_context.line_to(100, 100)
    aux_context.stroke()

    res = cairo.SVGSurface('test1.svg', 200, 200)
    context = cairo.Context(res)
    context.set_source_surface(aux, 0., 0.)
    context.paint()
    res.flush()
    res.finish()


def test2():
    aux = cairo.RecordingSurface(cairo.CONTENT_COLOR_ALPHA, None)
    aux_context = cairo.Context(aux)
    aux_context.move_to(0, 0)
    aux_context.line_to(100, 100)
    aux_context.stroke()

    res = cairo.SVGSurface('test2.svg', 200, 200)
    context = cairo.Context(res)
    context.set_source_surface(aux, 1., 2.)
    context.paint()
    res.flush()
    res.finish()
```

Results in

  - `test1.svg`
    ```svg
    <?xml version="1.0" encoding="UTF-8"?>
    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="200pt" height="200pt" viewBox="0 0 200 200" version="1.1">
    <g id="surface2">
    <path style="fill:none;stroke-width:2;stroke-linecap:butt;stroke-linejoin:miter;stroke:rgb(0%,0%,0%);stroke-opacity:1;stroke-miterlimit:10;" d="M 0 0 L 100 100 "/>
    </g>
    </svg>
    ```
  - `test2.svg`
    ```svg
    <?xml version="1.0" encoding="UTF-8"?>
    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="200pt" height="200pt" viewBox="0 0 200 200" version="1.1">
    <defs>
    <clipPath id="clip1">
      <rect width="0" height="-0"/>
    </clipPath>
    <g id="surface6" clip-path="url(#clip1)">
    </g>
    </defs>
    <g id="surface7">
    <use xlink:href="#surface6" transform="matrix(1,0,0,1,1,2)"/>
    </g>
    </svg>
    ```

So in the first case, the content of the `RecordingSurface` is written to the `SVGSurface`, but if you set a non-zero offset as in the second case, it isn't, though it does set the transformation matrix correctly.

If `PDFSurface` is substituted to `SVGSurface`, everything works as it should, so this issue is precisely with `SVGSurface`.

**Comment by Evpok** [comment]
Note that I also tried setting the pattern and painting it manually with the same results

```python
def test2():
    aux = cairo.RecordingSurface(cairo.CONTENT_COLOR_ALPHA, None)
    aux_context = cairo.Context(aux)
    aux_context.move_to(0, 0)
    aux_context.line_to(100, 100)
    aux_context.stroke()
    
    p = cairo.SurfacePattern(aux)
    p.set_matrix(cairo.Matrix(x0=-1., y0=-2.))

    res = cairo.SVGSurface('test2.svg', 200, 200)
    context = cairo.Context(res)
    context.set_source(p)
    context.rectangle(0, 0, 200, 200)
    context.fill()
    res.flush()
    res.finish()
```

**Comment by florianfesti** [comment]
I have a similar issue although I use pycairo-1.15.3 on Python 3. RecordingSurface on PDFSurface works just fine while it does not work at all with the SVGSurface. So it is quite possible that this is a cairo issue.

**Comment by liZe** [comment]
> I have a similar issue although I use pycairo-1.15.3 on Python 3. RecordingSurface on PDFSurface works just fine while it does not work at all with the SVGSurface. So it is quite possible that this is a cairo issue.

Definitely not a bug in cairocffi. We'd better open an issue on [Cairo's bugtracker](https://bugs.freedesktop.org/).

**Comment by Evpok** [comment]
Might be related to https://lists.freedesktop.org/archives/cairo/2017-October/028397.html I'll have to write at test for that

**Comment by Evpok** [comment]
Following up on this issue: setting bounds to the `RecordingSurface` fix this behaviour. Which confirms that this was due to the linked issue, which should be fixed in [cairo 1.15.10](https://www.cairographics.org/news/cairo-1.15.10/)

In the meantime, the following trick works

```python
def test2():
    aux = cairo.RecordingSurface(cairo.CONTENT_COLOR_ALPHA, None)
    aux_context = cairo.Context(aux)
    aux_context.move_to(0, 0)
    aux_context.line_to(100, 100)
    aux_context.stroke()

    aux2 = cairo.RecordingSurface(cairo.CONTENT_COLOR_ALPHA, aux.get_extents())
    auxcontext = cairo.Context(aux2)
    auxcontext.set_source_surface(aux, 0., 0.)
    auxcontext.paint()

    res = cairo.SVGSurface('test2.svg', 200, 200)
    context = cairo.Context(res)
    context.set_source_surface(aux2, 1., 2.)
    context.paint()
    res.flush()
    res.finish()
```

i.e. first work on an unbounded `RecordingSurface`, then create a bounded one of the right size and replay this one on the final `SVGSurface`.

**Comment by Evpok** [comment]
Epilogue : it was indeed a cairo-side issue that has been fixed in 1.15.10. Now I just have to wait for the Debian maintainer to package it :)

**Comment by Evpok** [comment]
Epilogue : it was indeed a cairo-side issue that has been fixed in 1.15.10. Now I just have to wait for the Debian maintainer to package it :)
